File: datasets/python_source-code/Multiple-B2/organic_7.py
# Two Sum (hash map) - organic submission 7
# Uses try/except KeyError instead of an 'in' membership check.

def two_sum(nums, target):
    seen = {}
    for i, num in enumerate(nums):
        complement = target - num
        try:
            match_index = seen[complement]
        except KeyError:
            seen[num] = i
            continue
        return [match_index, i]
    return None


if __name__ == "__main__":
    nums = [4, 6, 8, 10]
    target = 14
    print("Indices:", two_sum(nums, target))

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_b2/organic_7.java
// Two Sum (hash map) - organic submission 7
// Two-sum check checking null value lookup instead of containsKey.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            Integer matchIndex = seen.get(complement);
            if (matchIndex != null) {
                return new int[]{matchIndex, i};
            }
            seen.put(nums[i], i);
        }
        return null;
    }

    public static void main(String[] args) {
        int[] nums = {4, 6, 8, 10};
        int target = 14;
        System.out.println("Indices: " + Arrays.toString(twoSum(nums, target)));
    }
}
